src/bio.c

```diff
@@ -97,6 +97,7 @@ static int wolfSSL_BIO_BIO_read(WOLFSSL_BIO* bio, void* buf, int len)
 static int wolfSSL_BIO_MEMORY_read(WOLFSSL_BIO* bio, void* buf, int len)
 {
     int sz;
+    size_t bioSzThreshold = WOLFSSL_BIO_RESIZE_THRESHOLD;
```

Comment:
I don't really like this approach. I would prefer the intent to be explicit. Also the call to `wolfSSL_BUF_MEM_resize` might be dangerous in the future. Something like this would be better:
```
        if (bio->rdIdx >= bio->wrSz) {
            /* All data read */
            if (bio->flags & BIO_FLAGS_MEM_RDONLY) {
                /* Handle read-only case */
            }
            else {
                /* Handle resize case */
            }
        }
        else if (bio->rdIdx >= (int)bioSzThreshold &&
                !(bio->flags & BIO_FLAGS_MEM_RDONLY)) {
            /* Only resize when not read-only mode */
        }
```